component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -97,21 +98,18 @@ private static Map<String, String> generateClaims(String username, boolean refre
                 storedSecretKey = userClaimValues.get(secretKeyClaim);
                 if (StringUtils.isEmpty(storedSecretKey) || refresh) {
                     TOTPAuthenticatorKey key = generateKey(tenantDomain, context);
-                    generatedSecretKey = key.getKey();
-                    claims.put(secretKeyClaim, TOTPUtil.encrypt(generatedSecretKey));
+                    generatedSecretKey = key.getKey().toCharArray();
+                    claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, TOTPUtil.encrypt(
```

Comment:
In TOTPUtil.encrypt, we are passing the byte array to encryption. So, we can overload this method accept byte array and deprecate the existing method. 

https://github.com/wso2-extensions/identity-outbound-auth-totp/blob/5a1e04578ac39efe2f3b31997f1ff1aa549f2c2d/component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java#L82

Check if we can do the same for decrypt method as well